cmd/kubeadm/app/util/runtime/runtime_test.go

```diff
@@ -0,0 +1,301 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	kubeadmapiv1alpha3 "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1alpha3"
+	"k8s.io/utils/exec"
+	fakeexec "k8s.io/utils/exec/testing"
+)
+
+func TestNewContainerRuntime(t *testing.T) {
+	execLookPathOK := fakeexec.FakeExec{
+		LookPathFunc: func(cmd string) (string, error) { return "/usr/bin/crictl", nil },
+	}
+	execLookPathErr := fakeexec.FakeExec{
+		LookPathFunc: func(cmd string) (string, error) { return "", fmt.Errorf("%s not found", cmd) },
+	}
+	cases := []struct {
+		execer    fakeexec.FakeExec
+		criSocket string
+		isDocker  bool
+		isError   bool
+	}{
+		{execLookPathOK, kubeadmapiv1alpha3.DefaultCRISocket, true, false},
+		{execLookPathOK, "unix:///var/run/crio/crio.sock", false, false},
+		{execLookPathOK, "/var/run/crio/crio.sock", false, false},
+		{execLookPathErr, "unix:///var/run/crio/crio.sock", false, true},
+	}
+
+	for _, tc := range cases {
```

Comment:
Best practice here is to run each test case in a sub test. https://golang.org/pkg/testing/#T.Run

The output is much nicer and we know exactly which test case is failing. Right now the output would be confusing and we wouldn't know which of the test cases were failing.